extensions/dub-link-shortener/src/api/oauth.ts

```diff
@@ -0,0 +1,38 @@
+import { OAuth } from "@raycast/api";
+import { OAuthService } from "@raycast/utils";
+import { Dub } from "dub";
+import { workspaceApiKey } from "@utils/env";
+
+let dub: Dub | null = null;
+
+const client = new OAuth.PKCEClient({
+  redirectMethod: OAuth.RedirectMethod.Web,
+  providerName: "Dub",
+  providerIcon: "command-icon.png",
+  providerId: "dub",
+  description: "Connect your Dub account",
+});
+
+export const dubOAuth = new OAuthService({
+  client,
+  clientId: "dub_app_cf2e6f232bd9b1ca5918bc130b61ca127c2f1557ffcc95c4",
+  scope: ["workspaces.read", "links.write", "tags.read", "domains.read"].join(" "),
+  authorizeUrl: "https://app.dub.co/oauth/authorize",
+  tokenUrl: "https://api.dub.co/oauth/token",
+  refreshTokenUrl: "https://api.dub.co/oauth/token",
+  bodyEncoding: "url-encoded",
+  personalAccessToken: workspaceApiKey,
```

Comment:
We can totally just remove the workspaceKey from preferences if that is not required anymore and stick to OAuth. What do you think?

```suggestion
```

Reply:
We discussed this and decided we should support existing users with `workspaceKey` preferences. That way, users logged through a token don't have to authenticate again.

Reply:
I'm okay with removing it completely and prompt users to login with Dub, but maybe we want to keep it for backwards compatibility?

Reply:
I mean both approaches make sense to me. What I was thinking was if we remove it, then users will be prompted to logIn anyways (one-time) and that way we can manage the permissions in the code-base, rather than rely on users to define granular permissions and run into user issues. Especially when the downloads are low atm - that's what I did with Github Gist.

Reply:
So are we keeping both versions going forward @devkiran @steven-tey ?

Reply:
okay yeah sounds good to me – let's remove the workspaceId flow completely cc: @devkiran